### DISCUSSION TITLE
Plugins section in gradle are not being updated

### DISCUSSION THREAD

**Discussion by KordianD** [renovatebot/renovate#22362] [Request Help] [answered]
#### Was this something which used to work for you, and then stopped?

I never saw this working

#### Wanted end result.

Plugins section should be updated by renovatebot.

I cannot make `plugins` section in `build.gradle` updated.

This is small attempt to replicate the problem.

[repo](https://github.com/KordianD/renovatebot_test/blob/main/build.gradle)

I believe that `org.springframework.boot 2.7.10` should be updated to `org.springframework.boot 2.7.12`

**Comment by Churro** [comment, marked as answer]
After some diagnosis:
* [2.7.12](https://plugins.gradle.org/m2/org/springframework/boot/org.springframework.boot.gradle.plugin/) exists in https://plugins.gradle.org/m2 and is actually fetched by renovate among other releases
* Extraction works fine but the suggested update should be 2.7.12 instead of 3.1.0:

```
{
  "depType": "plugin",
  "depName": "org.springframework.boot",
  "packageName": "org.springframework.boot:org.springframework.boot.gradle.plugin",
  "commitMessageTopic": "plugin org.springframework.boot",
  "currentValue": "2.7.10",
  "managerData": {
    "fileReplacePosition": 103,
    "packageFile": "build.gradle"
  },
  "fileReplacePosition": 103,
  "datasource": "maven",
  "registryUrls": ["https://plugins.gradle.org/m2/"],
  "updates": [
    {
      "bucket": "latest",
      "newVersion": "3.1.0",
      "newValue": "3.1.0",
      "releaseTimestamp": "2023-05-22T19:48:03.000Z",
      "newMajor": 3,
      "newMinor": 1,
      "updateType": "major",
      "branchName": "renovate/all"
    }
  ],
  "warnings": [],
  "versioning": "gradle",
  "registryUrl": "https://plugins.gradle.org/m2",
  "currentVersion": "2.7.10",
  "isSingleVersion": true,
  "fixedVersion": "2.7.10"
}
```

~~I wonder if `:disableMajorUpdates` is applied too late here?~~

This is related to using the `group:all`, which includes `separateMajorMinor: false`, causing deps to be upgraded only to their latest release version. If used in combination with `:disableMajorUpdates`, you end up suppressing the update to 3.1.0, leaving you with no update at all.
